src/case-management/repositories/case-repository.js

```diff
@@ -0,0 +1,49 @@
+import { wreck } from '../../server/common/helpers/wreck.js'
+import { Case } from '../entities/case.js'
+import {
+  CaseRepositoryError,
+  CaseNotFoundError,
+  CaseUpdateError
+} from '../errors/case-errors.js'
+
+const toCase = (data) =>
+  new Case({
+    _id: data._id,
+    clientRef: data.payload?.clientRef || data.caseRef,
+    code: data.payload?.code,
+    workflowCode: data.workflowCode,
+    currentStage: data.currentStage,
+    stages: data.stages || [],
+    createdAt: data.payload?.createdAt || data.createdAt,
+    submittedAt: data.payload?.submittedAt || data.submittedAt,
+    status: data.status,
+    assignedUser: data.assignedUser
+  })
+
+export const findAll = async () => {
+  try {
+    const { payload } = await wreck.get('/cases')
+    const cases = payload.data || []
+    return cases.map(toCase)
+  } catch (error) {
+    throw new CaseRepositoryError('Failed to fetch cases')
+  }
```

Comment:
Can let it throw and add a catch all handler to render an error page based on Boom thrown by wreck.

e.g. https://github.com/DEFRA/cdp-node-frontend-template/blob/main/src/server/common/helpers/errors.js

hooked up as https://github.com/DEFRA/cdp-node-frontend-template/blob/bfe51452dc465817460b2c78d31168c1c53ef9dd/src/server/index.js#L65